unable to intialize alias type from fixed array

**V version: V 0.2.2 b09fa69.1b26ce1** 
**OS: macos, macOS, 11.4, 20F71** 

**What did you do?**
create an alias type of a fixed array, try to initalize a variable of the alias type.

```
// example 1
type Tres = [3]int
fixed_three := [1,2,3]!
x := Tres( fixed_three )

// example 2
x := Tres( [1,2,3]! )
```

**What did you expect to see?**
initalized variable x, containing the fixed three elements
 
**What did you see instead?**
```

                              ^
/tmp/v/ttt.8242516200375687969.tmp.c:10107:28: error: expected ';' after expression
        {main__Tres _ = (({1, 2, 3}));}
                                  ^
                                  ;
/tmp/v/ttt.8242516200375687969.tmp.c:10107:14: error: array initializer must be an initializer list or wide string literal
        {main__Tres _ = (({1, 2, 3}));}
                    ^
2 warnings and 3 errors generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -2512,9 +2512,12 @@ fn (mut g Gen) gen_assign_stmt(assign_stmt ast.AssignStmt) {
 			}
 			else {}
 		}
-		right_sym := g.table.get_type_symbol(g.unwrap_generic(val_type))
-		is_fixed_array_var := right_sym.kind == .array_fixed && (val is ast.Ident
-			|| val is ast.IndexExpr || val is ast.CallExpr
+		unwrapped_val_type := g.unwrap_generic(val_type)
+		right_sym := g.table.get_type_symbol(unwrapped_val_type)
+		unaliased_right_sym := g.table.get_final_type_symbol(unwrapped_val_type)
+		is_fixed_array_var := unaliased_right_sym.kind == .array_fixed && val !is ast.ArrayInit
+			&& (val is ast.Ident || val is ast.IndexExpr || val is ast.CallExpr
+			|| (val is ast.CastExpr && (val as ast.CastExpr).expr !is ast.ArrayInit)
 			|| val is ast.SelectorExpr)
 		g.is_assign_lhs = true
 		g.assign_op = assign_stmt.op
@@ -4319,6 +4322,8 @@ fn (mut g Gen) cast_expr(node ast.CastExpr) {
 		g.write('*(($styp *)(&')
 		g.expr(node.expr)
 		g.write('))')
+	} else if sym.kind == .alias && g.table.get_final_type_symbol(node.typ).kind == .array_fixed {
+		g.expr(node.expr)
 	} else {
 		styp := g.typ(node.typ)
 		mut cast_label := ''

--- a/vlib/v/tests/alias_fixed_array_init_test.v
+++ b/vlib/v/tests/alias_fixed_array_init_test.v
@@ -1,0 +1,12 @@
+type Tres = [3]int
+
+fn test_alias_fixed_array_init() {
+	fixed_three := [1, 2, 3]!
+	x := Tres(fixed_three)
+	println(x)
+	assert '$x' == 'Tres([1, 2, 3])'
+
+	y := Tres([2, 3, 4]!)
+	println(y)
+	assert '$y' == 'Tres([2, 3, 4])'
+}

